### PR TITLE
Bundle of changes (4-2-stable)

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

**Comment by trel** [comment]
why does moving it... oh... it was enforcing 'too often' / 'too deep'.   gotcha.

**Comment by korydraughn** [comment]
That is correct.

The admin check was being enforced on everything. I moved it so that the check is only done when a user invokes irule. For all other cases (i.e. delay / remote rules and PEPs firing via normal operation) we skip the admin check.

Ultimately, this change makes it so that only rodsadmins can invoke logical quotas rules directly.

Thoughts?

**Comment by alanking** [comment]
> Just for my own understanding... the privilege check for switching users is not needed here because it will be caught at a different level if it's not allowed to happen? What happens if a user which is not allowed to switch to some other user attempts to do so (or rather, the machinery tries to make that happen here, whether it was intended or not)?

Whoops, I meant to post this comment here. https://github.com/irods/irods_rule_engine_plugin_logical_quotas/issues/63#issuecomment-1065193897

**Comment by korydraughn** [comment]
Verified that all tests pass at the bench.

File: packaging/test_rule_engine_plugin_logical_quotas.py [hunk 1]
```diff
@@ -719,6 +719,29 @@ def test_group_owned_collections_do_not_require_the_admin_to_manually_change_acl
                 finally:
                     self.admin1.run_icommand(['iadmin', 'rmgroup', group_name])
 
+    @unittest.skipIf(test.settings.RUN_IN_TOPOLOGY, "Skip for Topology Testing")
+    def test_plugin_does_not_return_an_error_when_exec_rule_is_invoked_by_non_administrators__issue_63(self):
+        # This test verifies that regular users are allowed to invoke logical quotas rules indirectly.
+        # That is, a non-admin user will not be blocked by the REP unless they try to invoke a logical
```
Comment (alanking) [inline packaging/test_rule_engine_plugin_logical_quotas.py:746]: Should we add a test for the `irule` invocation? Is that even a thing that can be done?

Comment (korydraughn) [reply packaging/test_rule_engine_plugin_logical_quotas.py:746]: I thought about that and one already exists in the file.

Comment (alanking) [reply packaging/test_rule_engine_plugin_logical_quotas.py:746]: Very nice. Resolving, then!

**Review by alanking — APPROVED** [review]
Looks good to me

**Pull request merged**